monkey/infection_monkey/exploit/zerologon_utils/vuln_assessment.py

```diff
@@ -40,8 +43,15 @@ def is_exploitable(zerologon_exploiter_object) -> Tuple[bool, Optional[rpcrt.DCE
     # Connect to the DC's Netlogon service.
     try:
         rpc_con = zerologon_exploiter_object.connect_to_dc(zerologon_exploiter_object.dc_ip)
-    except Exception as e:
-        logger.info(f"Exception occurred while connecting to DC: {str(e)}")
+    except Exception as err:
+        error_message = f"Exception occurred while connecting to DC: {err}"
+        logger.info(error_message)
+        zerologon_exploiter_object._publish_exploitation_event(
+            target=zerologon_exploiter_object.host.ip_addr,
+            exploitation_success=False,
+            error_message=error_message,
+            tags=(ZEROLOGON_EXPLOITER_TAG, T1210_ATTACK_TECHNIQUE_TAG),
+        )
```

Comment:
Why are we publishing an event in here? Why not do it in the ZerologonExploiter class itself? It makes sense to me that this function's job is to check if the machine is exploitable, and that anything to do with the exploiter in the Agent, data gathered, events carried out, etc, is the job of the exploiter class.

Reply:
The thing is that every check of if a target is exploitable is a exploitation event so we need to record that. I didn't see any other way of doing it..

Reply:
I'm removing these from here for now.